content/post/20200426-kubernetes-the-universal-control-plane/index.md

```diff
@@ -0,0 +1,113 @@
+
+There is an obeservation that removing friction from an action leads to that action being done much more frequently. Kubernetes is headed to that same outcome with control planes. I think we will see control planes used to manage many more kinds of software. Systems are dynamic, and control loops are a powerful way to manage this reality. The bottleneck was always the implementation complexity, and Kubernetes is making moves to make this part tractable through Controllers and Operators paired with the Kubernetes Resource Model.
+
+## Standards Over Implementations
+
+<!-- Standardization = leverage and distribution; Implementing principles of programmable infrastructure and declarative APIs; KRM and controllers are core to the architecture, not married to the implementation; Kubernetes without computation is a schema database; Standardizing the core ideas will allow K8s to evolve the different infra needs across the industry -->
+The common theme threading through this essay is the idea of standardization. The Kubernetes Resource Model leans heavily on enforcing consistency in how resources are structured and tightly constrains the allowed access paths to the data. In return for the limitations on the data model [1] the ecosystem gains leverage, able to use the same primitives across many different abstractions. The primitives also implement the core principles to cloud computing: Resources are layered to create higher level abstractions (like how Deployments abstract Pods), and Controllers allow resources to be declarative by pushing the implementation complexity to the underlying code.
+
+If there any takeaways to be had, the idea of a universal control plane depends more on Kubernetes API design than container orchestration. Yes, without containers, Kubernetes is not much more than a database with a specific schema; It is the controllers that give resources meaning. But what makes the project more than a container orchestrator is that we can model abstractions by the data that defines it, then manage and change abstractions over time with consistent tooling and powerful control loops.
+
+Over time, we will see the Kubernetes project evolve over time in ways that may be suprising. I have my own ideas on what future Kubernetes will look like in my next essay, Kubernetes Unbundling. I will end this post with a question: What can we do with a universal control plane?
```

Comment:
I would change the last sentence to something like.

> I will leave you with a question: What exactly can we do now that we have the Universal Control Plane?

I prefer to avoid talking to the user like you're writing an essay and rather like you're there with them

Reply:
I would also change

> I have my own ideas on what future Kubernetes will look like in my next essay, Kubernetes Unbundling

into something like

> I will expand on my own ideas on what the future of Kubernetes will look like in a next post, Kubernetes Unbundling